Mathlib/Data/Set/Finite.lean

```diff
@@ -223,11 +236,14 @@ alias Finite.toFinset_ssubset_toFinset ↔ _ toFinset_strictMono
 -- Porting note: attribute [protected] doesn't work
 -- attribute [protected] toFinset_mono toFinset_strictMono
 
-@[simp]
+-- porting note: `simp` can simplify LHS but then it simplifies something
+-- in the generated `Fintype {x | p x}` instance and fails to apply `Set.toFinset_setOf`
+@[simp high]
 protected theorem toFinset_setOf [Fintype α] (p : α → Prop) [DecidablePred p]
     (h : { x | p x }.Finite) : h.toFinset = Finset.univ.filter p := by
   ext
-  simp
+  -- porting note: `simp` doesn't use the `simp` lemma `Set.toFinset_setOf` without the `_`
+  simp [Set.toFinset_setOf _]
```

Comment:
I pushed an uglier proof here in order to demonstrate what the comment above is talking about. It's not that `Set.toFinset_setOf` can't apply, it's just that `simp` is lousy at inferring arguments.